File: backend/models/Preco.cs
using System;

namespace backend.models
{
     public class Preco
     {
          public int Id { get; set; }
          public DateTime VigenciaInicial { get; set; }
          public DateTime VigenciaFinal { get; set; }
          public double ValorHora { get; set; }

          public Preco() { }

          public Preco(int id, DateTime vigenciaInicial, DateTime vigenciaFinal, double valorHora)
          {
               this.Id = id;
               this.VigenciaInicial = vigenciaInicial;
               this.VigenciaFinal = vigenciaFinal;
               this.ValorHora = valorHora;
          }
     }
}

File: frontend/src/app/models/Preco.ts
export class Preco {
  id?: number;
  vigenciaInicial: Date;
  vigenciaFinal: Date;
  valorHora: number;
  valorHoraAdicional: number;

  constructor(){
    this.id = 0;
    this.vigenciaInicial = null;
    this.vigenciaFinal = null;
    this.valorHora = 0.0;
    this.valorHoraAdicional = 0.0;
  }
}
